sdl_android/src/main/java/com/smartdevicelink/protocol/WiProProtocol.java

```diff
@@ -172,57 +172,59 @@ else if (protocolMsg.getBulkData() != null) {
 		}
 		
 		synchronized(messageLock) {
-			if (data.length > MAX_DATA_SIZE) {
-				
-				messageID++;
-	
-				// Assemble first frame.
-				int frameCount = data.length / MAX_DATA_SIZE;
-				if (data.length % MAX_DATA_SIZE > 0) {
-					frameCount++;
-				}
-				//byte[] firstFrameData = new byte[HEADER_SIZE];
-				byte[] firstFrameData = new byte[8];
-				// First four bytes are data size.
-				System.arraycopy(BitConverter.intToByteArray(data.length), 0, firstFrameData, 0, 4);
-				// Second four bytes are frame count.
-				System.arraycopy(BitConverter.intToByteArray(frameCount), 0, firstFrameData, 4, 4);
-
-				SdlPacket firstHeader = SdlPacketFactory.createMultiSendDataFirst(sessionType, sessionID, messageID, _version,firstFrameData,protocolMsg.getPayloadProtected());
-				firstHeader.setPriorityCoefficient(1+protocolMsg.priorityCoefficient);
-				//Send the first frame
-				handlePacketToSend(firstHeader);
-				
-				int currentOffset = 0;
-				byte frameSequenceNumber = 0;
-
-				for (int i = 0; i < frameCount; i++) {
-					if (i < (frameCount - 1)) {
-	                     ++frameSequenceNumber;
-	                        if (frameSequenceNumber ==
-	                                SdlPacket.FRAME_INFO_FINAL_CONNESCUTIVE_FRAME) {
-	                            // we can't use 0x00 as frameSequenceNumber, because
-	                            // it's reserved for the last frame
-	                            ++frameSequenceNumber;
-	                        }
-					} else {
-						frameSequenceNumber = SdlPacket.FRAME_INFO_FINAL_CONNESCUTIVE_FRAME;
-					} // end-if
-					
-					int bytesToWrite = data.length - currentOffset;
-					if (bytesToWrite > MAX_DATA_SIZE) { 
-						bytesToWrite = MAX_DATA_SIZE; 
+			if (data != null) {

```

Comment:
That was the intention of the fix, don't send anything if data is null. Clarifying question: what is the reason for the need to send the packet anyway if data is null? Is it because we need to keep track of the `messageId` field? Is there any other reason? Thanks.